beacon_node/http_api/src/block_id.rs

```diff
@@ -274,10 +274,27 @@ impl BlockId {
             warp_utils::reject::custom_not_found(format!("beacon block with root {}", root))
         })?;
 
+        // Error if the block is pre-Deneb and lacks blobs.
+        let blob_kzg_commitments = block.message().body().blob_kzg_commitments().map_err(|_| {
+            warp_utils::reject::custom_bad_request(
+                "block is pre-Deneb and has no blobs".to_string(),
+            )
+        })?;
+
         // Return the `BlobSidecarList` identified by `self`.
-        let blob_sidecar_list = chain
-            .get_blobs(&root)
-            .map_err(warp_utils::reject::beacon_chain_error)?;
+        let blob_sidecar_list = if !blob_kzg_commitments.is_empty() {
+            chain
+                .store
+                .get_blobs(&root)
+                .map_err(|e| warp_utils::reject::beacon_chain_error(e.into()))?
+                .ok_or_else(|| {
+                    warp_utils::reject::custom_not_found(format!(
```

Comment:
This looks like a good change to me. 

I'm not sure if we want to introduce new status codes, but there's also [`410 Gone`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/410) which would allow us to differentiate between block not found and expired blobs, without having to parse response message - but I'm also happy with 404.

In PeerDAS we'd also need a error response for nodes that are not capable of serving blobs data (custody columns < 64), and I _think_ we might be returning 404 too? 😅

Reply:
410 sounds good, but we would need to add it to the spec: https://ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/Beacon/getBlobSidecars.

Maybe we can advocate for that, while returning 404 in the meantime?

Reply:
Sounds good!